packages/portfolio-contract/src/schedule-order.ts

```diff
@@ -0,0 +1,91 @@
+type Ix = number;
+export type Job = {
+  taskQty: number;
+  /** entries of map from virtex to vertexes it depends on */
+  order: Array<[Ix, Ix[]]>;
+};
+
+const range = (n: number) => Array.from(Array(n).keys());
+
+/**
+ * ref [Promise.allSettled Return value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled#return_value)
+ */
+type PromiseSettledResult<T> =
+  | { status: 'fulfilled'; value: T }
+  | { status: 'rejected'; reason: any };
+
+const ok = {
+  status: 'fulfilled',
+  value: undefined,
+} as PromiseSettledResult<void>;
+harden(ok);
```

Comment:
`PromiseSettledResult` and `PromiseFulfilledResult` are already built into Typescript.


```suggestion
const ok = { status: 'fulfilled', value: undefined } as PromiseFulfilledResult<void>;
harden(ok);
```